### PR TITLE
Minor refactoring.

### PR DESCRIPTION
**Pull request by freezemage0**

### CODE REVIEW

File: Core/functions.php [hunk 1]
```diff
@@ -27,7 +27,7 @@ function env(): Env
 {
     $env = Env::get();
 
-    if ($env === null) {
+    if (!isset($env)) {
         throw new LogicException('Env is not set');
     }
```
Comment (CommandString) [inline Core/functions.php:32]: wouldn't comparing to null be faster cause there's no function call overhead, no?

File: Core/functions.php [hunk 2]
```diff
@@ -193,7 +189,7 @@ function getOptionFromInteraction(Collection|Interaction $options, string ...$na
             $options = $option?->options;
         }
 
-        if ($options === null || $option === null) {
```
Comment (CommandString) [inline Core/functions.php:196]: wouldn't comparing to null be faster cause there's no function call overhead, no?

File: Core/functions.php [hunk 3]
```diff
@@ -205,22 +201,22 @@ function getOptionFromInteraction(Collection|Interaction $options, string ...$na
...
 {
-    env()->discord->getLogger()->error($message, $context);
+    discord()?->getLogger()->error($message, $context);
 }
 
 function info(string $message, array $context = []): void
 {
-    env()->discord->getLogger()->info($message, $context);
+    discord()?->getLogger()->info($message, $context);
 }
```
Comment (CommandString) [inline Core/functions.php:220]: Calling env directly so there's not two function calls (I should've done `Env::get()` directly tbh)

Comment (freezemage0) [reply Core/functions.php:220]: Well, actually, I was thinking that `discord()` function should throw if `env()->discord` is null, but in the end I decided that this change is beyond the scope of current PR and left it out.

File: Core/functions.php (hunk 2 above)
Comment (freezemage0) [reply Core/functions.php:196]: Nanoseconds faster, probably.

If you want to optimize this, then there should be two `if`s with `break` in them. The first one should happen right after `$option` assignment within the loop that checks `$option === null` and `break`s the loop if true, while the other one should be after `$key !== count($names) - 1` check; it should check `if ($options === null)` and then `break`, if I understand the initial intent correctly.

Comment (freezemage0) [reply Core/functions.php:196]: Also `count($names) - 1` should be assigned to a variable before the `foreach ()` loop happens for an additional optimization.

File: Core/functions.php (hunk 1 above)
Comment (freezemage0) [reply Core/functions.php:32]: Yet again, nanoseconds faster. Also `isset` is not an actual function, it's a language construct, so it gets a different approach from the PHP interpreter and, iirc, it compiles into a different op code. Ultimately, using `isset()` is practically the same as strict comparison to `null`, but IMO the first one improves readability.

**Pull request merged**